tests/test_causal_model.py

```diff
@@ -275,119 +275,130 @@ def test_graph_input4(self, beta, num_instruments, num_samples, num_treatments):
         all_nodes = model._graph.get_all_nodes(include_unobserved=False)
         assert "Unobserved Confounders" not in all_nodes
 
-    @pytest.mark.parametrize(["beta", "num_instruments", "num_samples"],
-                             [(10, 2, 5000),])
-    def test_graph_refutation(self, beta, num_instruments, num_samples):
-        import numpy.random
-        numpy.random.seed(102)
-        data = dowhy.datasets.linear_dataset(beta=beta,
-        num_common_causes=5,
-        num_instruments = num_instruments,
-        num_effect_modifiers=1,
-        num_samples=num_samples, 
-        treatment_is_binary=True,
-        stddev_treatment_noise=10,
-        num_discrete_common_causes=1)
+    @pytest.mark.parametrize(["num_variables", "num_samples"],
+                             [(10,5000),])
+    def test_graph_refutation(self, num_variables,num_samples):
+        data = dowhy.datasets.dataset_from_random_graph(num_vars = num_variables, num_samples= num_samples)
         df = data["df"]
         model = CausalModel(
             data=df,
             treatment=data["treatment_name"],
             outcome=data["outcome_name"],
             graph=data["gml_graph"],
-            proceed_when_unidentifiable=True,
-            test_significance=None
         )
-        gml_str = """dag {
-        W0 [pos="-2.200,-1.520"]
-        W1 [pos="-1.457,-1.533"]
-        W2 [pos="-0.763,-1.547"]
-        W3 [pos="1.041,-1.587"]
-        W4 [pos="1.510,-1.560"]
-        X0 [pos="1.222,-0.625"]
-        Z0 [pos="0.390,-1.601"]
-        Z1 [pos="-0.176,-1.540"]
-        v0 [pos="-0.219,-0.881"]
-        y [pos="-0.144,-0.296"]
-        W0 -> v0
-        W0 -> y
-        W1 -> v0
-        W1 -> y
-        W2 -> v0
-        W2 -> y
-        W3 -> v0
-        W3 -> y
-        W4 -> v0
-        W4 -> y
-        X0 -> y
-        Z0 -> v0
-        Z1 -> v0
-        v0 -> y
-        }"""
-        model = CausalModel(
-            data=df,
-            treatment=data["treatment_name"],
-            outcome=data["outcome_name"],
-            graph=gml_str,
-            proceed_when_unidentifiable=True,
-            test_significance=None,
-            missing_nodes_as_confounders=True
-        )
-        graph_refutation_object = model.refute_graph(1,method_name = "partial_correlation")
-        assert graph_refutation_object.refutation_result == True
+        graph_refutation_object = model.refute_graph(k = 1, independence_test = 
+        {'test_for_continuous': 'partial_correlation', 
+        'test_for_discrete' : 'conditional_mutual_information'})
+        assert graph_refutation_object.number_of_constraints_satisfied >= 0.80 * graph_refutation_object.number_of_constraints_model
```

Comment:
Why 0.8? Why not 100% satisfied?

Reply:
Because given that 100% constraints are not supported, for now, that assertion would fail. I can change it to 100% if failing a test is okay.

Reply:
Ah, okay. I understand it better now. So here's a proposal:
1. Have the refute as None initially. For each (a,b,c), mark it True/False or "NotImplemented" if the test is not implemented yet. 
2. When you are aggregating the refutation result, mark it as Passed if there are not False tests. If there are non-zero "NotImplemented" tests, then refutationresult is True but issue a Warning saying that some tests could not be run because that config is not implemented. And if there are no False or "NotImplemented" tests, the result is True without any warnings. 

What do you think? Then you can change this line to check that False should zero. Both true and notimplemented are fine.

Reply:
We have a variable _results in GraphRefuter which has key as (a,b,c) and value as [p_value/cmi_value , True/False]. For unimplemented tests we can have value as [None, "Not_implemented"]

Reply:
yes, that sounds good